Sources/Helpers/AssociatedCellItemFactory.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  AssociatedCellItemFactory.swift
+//
+//  Copyright © 2019 Rosberry. All rights reserved.
+//
+
+import UIKit
+
+public class AssociatedCellItemFactory<U, T: UICollectionViewCell> {
+
+    /// Set this handler to retrieve a specific set of cell items for the associated object
+    ///
+    /// - Parameters:
+    ///    - Int: the index path of an object in the provided array
+    ///    - Any: the object associated with a cell item
+    public var initializationHandler: ((Int, U) -> [CollectionViewCellItem?])?
+    
+    /// Set this handler to configure the size of cell
+    ///
+    /// - Parameters:
+    ///    - Any: the object associated with a cell item
+    ///    - UICollectionView: collection view where cell should be placed
+    ///    - CollectionViewSectionItem: a section item in the section of which the cell should be placed
+    public var sizeConfigurationHandler: ((U, UICollectionView, CollectionViewSectionItem) -> CGSize)?
+    
+    /// Set this handler to configure the cell item
+    ///
+    /// - Parameters:
+    ///    - Any: the object associated with a cell item
+    ///    - CollectionViewCellItem: a cell item that should be cofigured
+    public var cellItemConfigurationHandler: ((Int, U, CollectionViewCellItem) -> Void)?
+    
+    /// Set this handler to configure the cell
+    ///
+    /// - Parameters:
+    ///    - Any: the object associated with a cell item
+    ///    - UICollectionViewCell: the cell that should be configured
+    ///    - CollectionViewCellItem: the cell item that performs a cell configuration
+    public var cellConfigurationHandler: ((U, T, CollectionViewCellItem) -> Void)?
+    
+    private var subfactories = [String: Any]()
```

Comment:
This variable doesn't have docs. Despite of it is private it still should be well-documented for better further support of the framework